hamcrest/src/main/java/org/hamcrest/text/IsBlankString.java

```diff
@@ -20,7 +20,16 @@ private IsBlankString() { }
 
     @Override
     public boolean matchesSafely(String item) {
-        return item.isBlank();
+        final int length = item.length();
+        int offset = 0;
+        while(offset < length) {
```

Comment:
Consider sticking to the Hamcrest code style, which has a space before `if` and opening parenthesis.
It's not enforced in the build apparently, but still...

Reply:
Can do. Are you using an autoformatter like spotless? It has Maven/Gradle plugins. Might be worth pulling into a separate PR :)

Reply:
Thanks for fixing.
I was talking about a checker rather than a formatter.
JavaHamcrest is already using CheckStyle, but apparently it doesn't check much besides the fact that there are no tabs.
For the record: I'm not an official contributor to JavaHamcrest, I'm just a random someone with a stake in the project like you, I don't have the power to merge this PR even if I wanted to.

I imagine you inspired your implementation on Java 11 JDK's sources?
They look similar, yet a little bit different, and I wonder why...

The JDK's implementation distinguishes between Latin-1 and UTF-16 encoded strings. From what I can tell a `String` can be encoded in either of these 2. Your implementation only reflects the JDK's UTF-16 implementation. Is it safe to assume UTF-16? Does that also cover Latin-1? If such an assumption were safe, why does the JDK's implementation bother to make a difference?

In principle, I agree with this change, I'm just wondering if we should hold off fixing until JavaHamcrest bumps source/target compatibility to Java 11, in which case the fix will become simpler (i.e. your first commit).